### PR TITLE
HttpSyncerTest Kotlin Migration

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Kotlin Migration

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt [hunk 1]
```diff
@@ -14,129 +14,106 @@
...
-    public void getDefaultMediaUrlWithNoHostNum() {
-        HttpSyncer underTest = getServerWithHostNum(null);
-
-        String syncUrl = underTest.syncURL();
-
-        assertThat(syncUrl, is(sDefaultUrlNoHostNum));
+    fun defaultMediaUrlWithNoHostNum() {
+        val underTest = getServerWithHostNum(null)
+        val syncUrl = underTest.syncURL()
+        MatcherAssert.assertThat(syncUrl, Matchers.`is`(sDefaultUrlNoHostNum))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt:35]: re-import `MatcherAssert` and `Matchers`

File: AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt [hunk 2]
```diff
@@ -14,129 +14,106 @@
...
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.ichi2.anki.AnkiDroidApp
+import com.ichi2.utils.KotlinCleanup
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.runner.RunWith
 
+@RunWith(AndroidJUnit4::class)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt:28]: cleanup: `is` => `equalTo`

**Review by david-allison — COMMENTED** [review]
Very nice, two nits, then good to go

File: AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt [hunk 3]
```diff
@@ -14,129 +14,107 @@
...
+        val syncUrl = underTest.syncURL()
+        assertThat(syncUrl, `is`(sDefaultUrlWithHostNum))
     }
 
-    private void setCustomServerWithNoUrl() {
-        SharedPreferences userPreferences = AnkiDroidApp.getSharedPrefs(AnkiDroidApp.getInstance());
-        userPreferences.edit().putBoolean("useCustomSyncServer", true).apply();
+    private fun setCustomServerWithNoUrl() {
+        val userPreferences = AnkiDroidApp.getSharedPrefs(AnkiDroidApp.getInstance())
+        userPreferences.edit().putBoolean("useCustomSyncServer", true).apply()
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt:98]: Add a kotlin cleanup to use the `edit{}` extension function on SharedPreferences.Editor.

File: AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt [hunk 4]
```diff
@@ -14,129 +14,107 @@
...
-        e.apply();
+    private fun getServerWithHostNum(hostNum: Int?): HttpSyncer {
+        return HttpSyncer(null, null, HostNum(hostNum))
     }
 
-    @NonNull
-    private HttpSyncer getServerWithHostNum(Integer hostNum) {
-        return new HttpSyncer(null, null, new HostNum(hostNum));
+    companion object {
+        private const val sCustomServerWithNoFormatting = "https://sync.example.com/"
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/libanki/sync/HttpSyncerTest.kt:115]: Add a cleanup to rename this constants.

**Review by lukstbit — CHANGES_REQUESTED** [review]

**Review by lukstbit — APPROVED** [review]

**Review by ShridharGoel — APPROVED** [review]

**Comment by divyansh-dxn** [comment]
I mistakenly had pushed more commits to this branch, but I dropped them instantly.

**Pull request merged**